File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] rank = {6, 6, 5, 4, 3, 2, 1};
        
        int match_cnt = 0; int zero_cnt = 0;
        for (int lotto : lottos) {
            if (lotto == 0) {
                zero_cnt += 1;
                continue;
            }
            for (int num : win_nums) {
                if (lotto == num) {
                    match_cnt += 1;
                }
            }
        }
        
        int[] answer = {rank[match_cnt + zero_cnt], rank[match_cnt]};
        return answer;
    }
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    rank = [6, 6, 5, 4, 3, 2, 1]
    
    match_cnt, zero_cnt = 0, lottos.count(0)
    for n in win_nums:
        match_cnt += lottos.count(n)
    return [rank[match_cnt + zero_cnt], rank[match_cnt]]